nsis/src/nsis/NsisLoader.java

```diff
@@ -144,21 +171,30 @@ public Data createData(Program program, Listing listing, Address address,
 		return null;
 	}
 
-	private void processBlockHeaders(Program program, TaskMonitor monitor,
-			BinaryReader reader, long nsis_header_offset) {
+	/**
+	 * Initializes the block headers and adds them to the "Program Trees" view
+	 * in Ghidra.
+	 * 
+	 * @param program
+	 * @param reader
+	 * @param startingAddr, the Address where the nsis script header starts
+	 * @throws IOException
+	 */
+	private void initBlockHeaders(Program program, BinaryReader reader,
+			Address startingAddr) throws IOException {
 		int block_header_offset = NsisScriptHeader.getHeaderSize();
 		for (int i = 0; i < NsisConstants.NB_NSIS_BLOCKS; i++) {
 			System.out.printf("Processing block at offset %08x\n",
-					block_header_offset);
-			ghidra.program.model.address.Address block_address;
-			block_address = program.getAddressFactory().getDefaultAddressSpace()
-					.getAddress(block_header_offset);
+					block_header_offset + startingAddr.getOffset());
+			Address block_address = startingAddr.add(block_header_offset);
+
+			reader.setPointerIndex(
+					startingAddr.getOffset() + block_header_offset);
 
-			reader.setPointerIndex(nsis_header_offset + block_header_offset);
 			NsisBlockHeader block_header = new NsisBlockHeader(reader);
 			System.out.printf("Block index: %d\n", i);
```

Comment:
Not for this PR: Ghidra has a logging system, we should probably convert the System.out.println to use the Ghidra logging system.

Reply:
Good to know, I'll add this in my next PRs

Reply:
ref: #15